src/ui_utils.c

```diff
@@ -2440,20 +2441,32 @@ void ui_init_builder(void)
 	gtk_builder_set_translation_domain(builder, GETTEXT_PACKAGE);
 
 	error = NULL;
-	interface_file = g_build_filename(app->datadir, "geany.glade", NULL);
-	if (! gtk_builder_add_from_file(builder, interface_file, &error))
+	ui_data = g_resource_lookup_data(geany_get_resource(),
+		"/org/geany/Geany/geany.glade", G_RESOURCE_LOOKUP_FLAGS_NONE, &error);
+	if (ui_data == NULL)
+	{
+		dialogs_show_msgbox_with_secondary(GTK_MESSAGE_ERROR,
+			_("Geany cannot start!"), error->message);
+		g_error("Cannot load user-interface: %s", error->message);
+		g_error_free(error);
+		g_object_unref(builder);
+		return;
+	}
+
+	error = NULL;
+	if (gtk_builder_add_from_string(builder, g_bytes_get_data(ui_data, NULL),
+		g_bytes_get_size(ui_data), &error) == 0)
```

Comment:
You could keep the boolean style condition (`if (gtk_builder_...())`) instead of checking for `0`; it's the working the same yet is a little more straightforward in a boolean-filled world, and the use of `guint` really seem like an API glitch in the GTK side, as it's just documented to return "A positive value on success, 0 if an error occurred".

Reply:
I like the explicit `== 0` since it shows that the result is not semantically a boolean.

Reply:
Well, the thing is that from what I understand from the API, it actually has the semantics of a boolean, it seems like the actual value has literally *no* meaning apart from success/failure.  I didn't check the code yet, but AFAICT the documentation doesn't say anything about any use for it, and there's no API that would use the returned value.  But anyway, it doesn't really matter to me either way.

Reply:
Without any actual research, I assumed it's related to GtkUIManager "merge id" return value, but I have no good evidence that is the case.

Reply:
Compiler will almost certainly generate the same code either way, but maybe we should check for 666 in case GTK is possessed :)

Reply:
@codebrainz interesting idea, but no: https://git.gnome.org/browse/gtk+/tree/gtk/gtkbuilder.c#n1299

Reply:
So why isn't the return type `gboolean`? It's the same width and conveys the semantics better.